util/chunk/chunk.go

```diff
@@ -59,32 +106,29 @@ func (c *Chunk) MemoryUsage() (sum int64) {
 	return
 }
 
-// addFixedLenColumn adds a fixed length column with elemLen and initial data capacity.
-func (c *Chunk) addFixedLenColumn(elemLen, initCap int) {
-	c.columns = append(c.columns, &column{
+// newFixedLenColumn creates a fixed length column with elemLen and initial data capacity.
+func newFixedLenColumn(elemLen, initCap int) *column {
+	return &column{
 		elemBuf:    make([]byte, elemLen),
 		data:       make([]byte, 0, initCap*elemLen),
 		nullBitmap: make([]byte, 0, initCap>>3),
-	})
+	}
 }
 
-// addVarLenColumn adds a variable length column with initial data capacity.
-func (c *Chunk) addVarLenColumn(initCap int) {
-	c.columns = append(c.columns, &column{
+// newVarLenColumn creates a variable length column with initial data capacity.
+func newVarLenColumn(initCap int, old *column) *column {
+	estimatedElemLen := 8
+	// For varLenColumn(e.g. varchar) we could not take the accuracy length of element,
+	// so, in first executor.Next we using a experience value --- 8(so it maybe make `runtime.growslice`)
```

Comment:
Therefore, in the first executor.Next we use an experience value -- 8 (so it may make `runtime.growslice`)